src/coreclr/nativeaot/System.Private.CoreLib/src/System/Diagnostics/Debugger.cs

```diff
@@ -74,5 +76,11 @@ public static bool IsLogging()
             }
             return false;
         }
+
+        internal static bool IsAnyDebuggerAttached() => IsAnyDebuggerAttachedInternal() != 0;
+
+        [LibraryImport(RuntimeImports.RuntimeLibrary, EntryPoint = "DebugDebugger_IsAnyDebuggerAttached")]
+        [SuppressGCTransition]
```

Comment:
The Unix implementations of minipal_is_native_debugger_present do not look very light weight. Is SuppressGCTransition really appropriate here?

Reply:
`minipal_is_native_debugger_present` performs a syscall on OSX and reads a procfs file on Linux. Does this preclude it from being a candidate for `SuppressTransition`?

Reply:
`SuppressGCTransition` should be only used in places where it makes a difference. Does it make the check significantly faster?

Reply:
Thanks, yes I don't think it is making any difference, I just added it while addressing another feedback.